### PR TITLE
fix: Linter fixes for plugins/inputs/[a-o]* (leftovers)

### PR DESCRIPTION
**Pull request by zak-pawel**
Following findings were fixed:
```
plugins/inputs/dcos/client.go:330:19                                       staticcheck  SA1019: jwt.StandardClaims is deprecated: Use RegisteredClaims instead for a forward-compatible way to access registered claims in a struct. 
plugins/inputs/dcos/dcos.go:240:3                                          gosimple     S1017: should replace this `if` statement with an unconditional `strings.TrimPrefix`
plugins/inputs/execd/shim/goshim_posix.go:18:3                             gosimple     S1000: should use a simple channel send/receive instead of `select` with a single case
plugins/inputs/execd/shim/shim_test.go:124:2                               revive       import-shadowing: The name 'inputs' shadows an import name
plugins/inputs/kafka_consumer/kafka_consumer.go:162:41                     revive       import-shadowing: The name 'config' shadows an import name
plugins/inputs/kafka_consumer/kafka_consumer.go:167:62                     revive       import-shadowing: The name 'config' shadows an import name
plugins/inputs/kafka_consumer/kafka_consumer.go:194:2                      revive       import-shadowing: The name 'config' shadows an import name
plugins/inputs/kafka_consumer/kafka_consumer_test.go:46:62                 revive       import-shadowing: The name 'config' shadows an import name
plugins/inputs/minecraft/internal/rcon/rcon.go:103:4                       revive       bare-return: avoid using bare returns, please add return expressions
plugins/inputs/minecraft/internal/rcon/rcon.go:107:2                       nakedret     naked return in func `Authorize` with 12 lines of code
plugins/inputs/minecraft/internal/rcon/rcon.go:216:2                       nakedret     naked return in func `NewClient` with 6 lines of code
plugins/inputs/minecraft/internal/rcon/rcon.go:70:3                        revive       bare-return: avoid using bare returns, please add return expressions
plugins/inputs/minecraft/internal/rcon/rcon.go:72:3                        revive       bare-return: avoid using bare returns, please add return expressions
plugins/inputs/minecraft/internal/rcon/rcon.go:74:3                        revive       bare-return: avoid using bare returns, please add return expressions
plugins/inputs/minecraft/internal/rcon/rcon.go:98:3                        revive       early-return: if c {...} else {... return } can be simplified to if !c { ... return } ...
plugins/inputs/opcua/opcua_util.go:12:2                                    revive       imports-blacklist: should not use the following blacklisted import: "log"
```

Also all `assert.` for these plugins were replaced by `require.`

### CODE REVIEW

File: plugins/inputs/dcos/client.go [hunk 1]
```diff
@@ -100,7 +100,7 @@ type ClusterClient struct {
 
 type claims struct {
 	UID string `json:"uid"`
-	jwt.StandardClaims
+	jwt.RegisteredClaims
```
Comment (srebhan) [inline plugins/inputs/dcos/client.go:103]: Why is this change needed? Doesn't that change the JSON source/target?

File: plugins/inputs/minecraft/internal/rcon/rcon.go [hunk 2]
```diff
@@ -67,17 +67,17 @@ func (p Packet) Compile() (payload []byte, err error) {
...
 	}
 
-	if _, err := buffer.WriteString(p.Body); err != nil {
+	if _, err = buffer.WriteString(p.Body); err != nil {
 		return nil, err
 	}
-	if _, err := buffer.Write(padding[:]); err != nil {
+	if _, err = buffer.Write(padding[:]); err != nil {
 		return nil, err
 	}
```
Comment (srebhan) [inline plugins/inputs/minecraft/internal/rcon/rcon.go:82]: You probably should use local errors with `if err := ...; err != nil {` here concisely?

File: plugins/inputs/opcua/opcua_client.go [hunk 3]
```diff
@@ -476,7 +475,7 @@ func (o *OpcUA) setupOptions() error {
 		}
 	}
 
-	o.opts, err = generateClientOpts(endpoints, o.Certificate, o.PrivateKey, o.SecurityPolicy, o.SecurityMode, o.AuthMethod, o.Username, o.Password, time.Duration(o.RequestTimeout))
+	o.opts, err = o.generateClientOpts(endpoints, o.Certificate, o.PrivateKey, o.SecurityPolicy, o.SecurityMode, o.AuthMethod, o.Username, o.Password, time.Duration(o.RequestTimeout))
```
Comment (srebhan) [inline plugins/inputs/opcua/opcua_client.go:478]: If you are changing this anyway, you should maybe get rid of the struct-owned function parameters...

**Review by srebhan — COMMENTED** [review]
Thanks @zak-pawel for cleaning up! I have some comments, the most critical one is the renaming of the `jwt` stuct field. Doesn't that change the JSON layout and thus has an unintented effect?

File: plugins/inputs/dcos/client.go (hunk 1 above)
Comment (zak-pawel) [reply plugins/inputs/dcos/client.go:103]: Because `StandardClaims` is deprecated.

From: `github.com/golang-jwt/jwt/v4@v4.1.0/claims.go`:
```
// StandardClaims are a structured version of the JWT Claims Set, as referenced at
// https://datatracker.ietf.org/doc/html/rfc7519#section-4. They do not follow the
// specification exactly, since they were based on an earlier draft of the
// specification and not updated. The main difference is that they only
// support integer-based date fields and singular audiences. This might lead to
// incompatibilities with other JWT implementations. The use of this is discouraged, instead
// the newer RegisteredClaims struct should be used.
//
// Deprecated: Use RegisteredClaims instead for a forward-compatible way to access registered claims in a struct.
type StandardClaims struct {
	Audience  string `json:"aud,omitempty"`
	ExpiresAt int64  `json:"exp,omitempty"`
	Id        string `json:"jti,omitempty"`
	IssuedAt  int64  `json:"iat,omitempty"`
	Issuer    string `json:"iss,omitempty"`
	NotBefore int64  `json:"nbf,omitempty"`
	Subject   string `json:"sub,omitempty"`
}

...

// RegisteredClaims are a structured version of the JWT Claims Set,
// restricted to Registered Claim Names, as referenced at
// https://datatracker.ietf.org/doc/html/rfc7519#section-4.1
//
// This type can be used on its own, but then additional private and
// public claims embedded in the JWT will not be parsed. The typical usecase
// therefore is to embedded this in a user-defined claim type.
//
// See examples for how to use this with your own claim types.
type RegisteredClaims struct {
	// the `iss` (Issuer) claim. See https://datatracker.ietf.org/doc/html/rfc7519#section-4.1.1
	Issuer string `json:"iss,omitempty"`
	// the `sub` (Subject) claim. See https://datatracker.ietf.org/doc/html/rfc7519#section-4.1.2
	Subject string `json:"sub,omitempty"`
	// the `aud` (Audience) claim. See https://datatracker.ietf.org/doc/html/rfc7519#section-4.1.3
	Audience ClaimStrings `json:"aud,omitempty"`
	// the `exp` (Expiration Time) claim. See https://datatracker.ietf.org/doc/html/rfc7519#section-4.1.4
	ExpiresAt *NumericDate `json:"exp,omitempty"`
	// the `nbf` (Not Before) claim. See https://datatracker.ietf.org/doc/html/rfc7519#section-4.1.5
	NotBefore *NumericDate `json:"nbf,omitempty"`
	// the `iat` (Issued At) claim. See https://datatracker.ietf.org/doc/html/rfc7519#section-4.1.6
	IssuedAt *NumericDate `json:"iat,omitempty"`
	// the `jti` (JWT ID) claim. See https://datatracker.ietf.org/doc/html/rfc7519#section-4.1.7
	ID string `json:"jti,omitempty"`
}
```

Code uses only `ExpiresAt` field which for both cases has `json:"exp,omitempty"` tag. 

In my opinion this change is ok, but would love to have 2nd person to check this out carefully.

File: plugins/inputs/minecraft/internal/rcon/rcon.go (hunk 2 above)
Comment (zak-pawel) [reply plugins/inputs/minecraft/internal/rcon/rcon.go:82]: I can, but since there is named return "argument" called `err`, it can be reused everywhere. Or not?

File: plugins/inputs/opcua/opcua_client.go (hunk 3 above)
Comment (zak-pawel) [reply plugins/inputs/opcua/opcua_client.go]: Sure, done!

File: plugins/inputs/dcos/client.go (hunk 1 above)
Comment (srebhan) [reply plugins/inputs/dcos/client.go:103]: Ok I see. As only the expire data is used and this one is set directly with your change, I agree that the change is fine.

File: plugins/inputs/minecraft/internal/rcon/rcon.go (hunk 2 above)
Comment (srebhan) [reply plugins/inputs/minecraft/internal/rcon/rcon.go:82]: I agree.

**Review by srebhan — APPROVED** [review]
Looks good to me.

**Review by sspaink — APPROVED** [review]
